File: #03/Categoria 03/q03 - iguais/iguais.js
iguais = (a,b,c) => {
    if (a == b && a == c) {
        return 3
    } else if ((a == b) || (b == c) || (a == c)) {
        return 2
    } else {
        return 0
    }
}

console.log(iguais(1,2,3))
console.log(iguais(1,1,3))
console.log(iguais(1,1,1))

File: #03/Categoria 03/q03 - iguais/iguais.hs
main :: IO ()
main = do
    a <- readLn :: IO Int
    b <- readLn :: IO Int
    c <- readLn :: IO Int
    print $ iguais a b c

iguais :: (Eq a, Num p) => a -> a -> a -> p
iguais a b c
  | (a == b) && (a == c) = 3
  | (a == b) || (b == c) || (a == c) = 2
  | otherwise = 0